modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+
+A component for making virtualized tables.
+
+[source,ts]
+----
+const MachineList: React.FC<MachineListProps> = (props) => {
+  return (
+    <VirtualizedTable<MachineKind>
+     {...props}
+     aria-label='Machines'
+     columns={getMachineColumns}
+     Row={getMachineTableRow}
+    />
+  );
+}
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`data` |data for table
+|`loaded` |flag indicating data is loaded
+|`loadError` |error object if issue loading data
+|`columns` |column setup
+|`Row` |row setup
+|`unfilteredData` |original data without filter
+|`NoDataEmptyMsg` |(optional) no data empty message component
+|`EmptyMsg` |(optional) empty message component
+|`scrollNode` |(optional) function to handle scroll
+|`label` |(optional) label for table
+|`ariaLabel` |(optional) aria label
+|`gridBreakPoint` |sizing of how to break up grid for responsiveness
+|`onSelect` |(optional) function for handling select of table
+|`rowData` |(optional) data specific to row
+|===
+
+[discrete]
+== `TableData`
+
+Component for displaying table data within a table row.
+
+[source,ts]
+
+----
+const PodRow: React.FC<RowProps<K8sResourceCommon>> = ({ obj, activeColumnIDs }) => {
+  return (
+    <>
+      <TableData id={columns[0].id} activeColumnIDs={activeColumnIDs}>
+        <ResourceLink kind="Pod" name={obj.metadata.name} namespace={obj.metadata.namespace} />
+      </TableData>
+      <TableData id={columns[1].id} activeColumnIDs={activeColumnIDs}>
+        <ResourceLink kind="Namespace" name={obj.metadata.namespace} />
+      </TableData>
+    </>
+  );
+};
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`id` |unique id for table
+|`activeColumnIDs` |active columns
+|`className` |(optional) option class name for styling
+|===
+
+[discrete]
+== `useActiveColumns`
+
+A hook that provides a list of user-selected active TableColumns.
+
+[source,tsx]
+----
+  // See implementation for more details on TableColumn type
+  const [activeColumns, userSettingsLoaded] = useActiveColumns({
+    columns,
+    showNamespaceOverride: false,
+    columnManagementID,
+  });
+  return userSettingsAreLoaded ? <VirtualizedTable columns={activeColumns} {...otherProps} /> : null
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`options` |Which are passed as a key-value map
+
+|`` |\{TableColumn[]} options.columns - An array of all available
+TableColumns
+
+|`` |\{boolean} [options.showNamespaceOverride] - (optional) If true, a
+namespace column will be included, regardless of column management
+selections
+
+|`` |\{string} [options.columnManagementID] - (optional) A unique id
+used to persist and retrieve column management selections to and from
+user settings. Usually a 'group~verion~kind' string for a resource.
+|===
+
+A tuple containing the current user selected active columns (a subset of options.columns), and a boolean flag indicating whether user settings have been loaded.
```

Comment:
s/A tuple/It returns a tuple 

s/current user selected/current user-selected or current-user selected ??